fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -107,6 +108,16 @@ private FDBDirectoryMergeScheduler(@Nonnull IndexMaintainerState state, int merg
         })
         @Override
         public synchronized void merge(final MergeSource mergeSource, final MergeTrigger trigger) throws IOException {
+            if (trigger == MergeTrigger.FULL_FLUSH || trigger == MergeTrigger.CLOSING) {
+                final IndexDeferredMaintenancePolicy indexDeferredMaintenancePolicy = state.store.getIndexDeferredMaintenancePolicy();
+                if (!indexDeferredMaintenancePolicy.shouldAutoMergeDuringCommit()) {
+                    // If this store is being closed, it is too late to use it as a messenger. This flag should also be set
+                    // when the requirements for a "real" merge are met.
+                    indexDeferredMaintenancePolicy.setMergeRequired(true);
```

Comment:
Here we need to "queue" a deferred merge operation. I don't know whether that's literally some kind of operation state or just names / flags.